.github/actions/clang-tidy/ament_clang_tidy.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import json
+import os
+import re
+import subprocess
+import sys
+import time
+from collections import defaultdict
+from multiprocessing.pool import ThreadPool
+from xml.sax.saxutils import escape, quoteattr
+
+import yaml
+
+
+def main(argv=sys.argv[1:]):
+    extensions = ["c", "cc", "cpp", "cxx", "h", "hh", "hpp", "hxx"]
+    excluded_files = [
```

Comment:
Hard-coding excluded files here makes it obscure and harder to update.

Reply:
Yea this was just to get something working, can create a .ament-clang-tidy-ignore file that works like .gitignore

Reply:
I've written up a solution that follows what it does for gtest. Give me a minute to push it.

Reply:
I also realized I put the function in the wrong place, can you move its definition to immediately below `is_unittest_source`